Nasbench201

PhMueller opened the pull request:
Consecutive Pull Request after #36. 

This pull request contains the code for the nasbench201 benchmark.

The data for this benchmark will be downloaded by the start of the benchmark. The download function is thread-safe!

PhMueller committed:
MOVE Lock files to cache folder + FIX test cases Part 1

PhMueller committed:
TRAVIS should crash if an error occurs.

PhMueller committed:
FIX test cases Part 2

PhMueller committed:
FIX test cases Part 3

PhMueller committed:
ADD codecov.yml

PhMueller committed:
ADD test cases

PhMueller committed:
REMOVE test_data_manager.py

Uploaded a file for a future pull request

PhMueller committed:
ADD Nasbench201

PhMueller committed:
FIX paths in recipe

PhMueller committed:
ADD small util function to activate / deactivate debugging on container side

PhMueller committed:
NASBench201 test should partially use the container

PhMueller committed:
NASBench201 takes List as data seed.

This is because we send the payload as json to the container. And tuples are casted to lists by json.

PhMueller committed:
FIX ConfigSpace is not deterministic via container

The CS was not deterministic since the seed was not set after receiving the cs string from the contianer.

PhMueller committed:
FIX dependencies for nasbench201

PhMueller committed:
FIX dependencies for nasbench201

PhMueller committed:
ADD Singularity dependencies to standard dependencies

We need it everywhere, so I've removed the singularity tag and add the pyro and the oslo dependency to the default requirements.

PhMueller committed:
[skip travis] Update README.md

Commit:
Merge branch 'development' into Nasbench201

KEggensperger commented on hpolib/benchmarks/nas/nasbench_201.py:
```diff
@@ -0,0 +1,445 @@
+        eval_times = [self.data[(seed, 'eval_times')][structure_str][epoch] for seed in data_seed]
+
+        return {'function_value': float(100 - np.mean(train_accuracies)),
+                'cost': float(np.mean(train_times) * epoch),
```
Could you please return here the actual sum of the training epochs? I realized that there are indeed different runtimes per configuration per epoch (they only slightly vary, but they vary).

Also, if you return the mean across N seeds, the runtime should also reflect the total runtime of training N networks

PhMueller committed:
[skip travis] nasbench_201 returns now the sum of the time per epochs as cost

- Imporve the doc strings. Explain the cost computation + fix a mistake in the used identifiers
- the nasbench201 returns now the exact sum of time needed to train a network for x epochs.
- the evaluation costs in the objective function is now train cost + eval cost. So that it is equal to the return value of the objective_function_test.
- Minor improvement in the logging. Moved the logging from the benchmark inti in the correct datamanager class.

PhMueller committed:
Merge remote-tracking branch 'origin/Nasbench201' into Nasbench201

PhMueller committed:
[skip travis] Update test cases

PhMueller committed:
Trigger Travis
